src/blog/2024/06/interacting-with-google-sheet-from-node-red.md

```diff
@@ -0,0 +1,150 @@
+
+10. Click "Done." Make sure to copy the generated service account email and save it for later use.
+
+!["Screenshot showing the 'Done' button"](./images/interacting-with-google-sheet-from-node-red-delete-page_11.png "Screenshot showing the 'Done' button"){data-zoomable}
+
+!["Screenshot showing the created service account email"](./images/interacting-with-google-sheet-from-node-red-delete-page_14.png "Screenshot showing the created service account email"){data-zoomable}
+
+11. To generate a private key, click on the three dots icon on the right of the newly created service account and select "Manage keys."
+
+!["Screenshot showing the three dot icon"](./images/interacting-with-google-sheet-from-node-red-delete-page_15.png "Screenshot showing the three dot icon"){data-zoomable}
+
+12. Click on "Add key," choose "Create new key," select "JSON" as the key type, and click "Create." Your private key will be generated and downloaded. 
+
+!["Screenshot showing the 'Add key' and the 'Create new key'"](./images/interacting-with-google-sheet-from-node-red-delete-page_15.png "Screenshot showing the 'Add key' and the 'Create new key'"){data-zoomable}
+
+!["Screenshot showing the 'JSON' option and 'Create new key' button"](./images/interacting-with-google-sheet-from-node-red-delete-page_13.png "Screenshot showing the 'JSON' option and 'Create new key' button"){data-zoomable}
+
+### Configuring the Google Sheet Node
+
+Before proceeding, ensure you have added the [environment variable](/blog/2023/01/environment-variables-in-node-red/) for the private key that was generated. Additionally, grant the editor access to the sheet you want to interact with for that service account email we created in the above section.
+
+1. Drag a GSheet node onto the canvas.
+2. Double-click on the node and click on the pencil icon next to "creds."
+3. Enter the environment variable added for the private key in the "creds" field and click "Add."
+4. Go to the Google Sheet you want to interact with and copy its ID from the URL. The URL will be in this format: `https://docs.google.com/spreadsheets/d/<id_of_sheet>/`
+5. Return to your Node-RED instance, double-click on the GSheet node again, and paste the spreadsheet ID into the "SpreadsheetID" field.
+6. Enter the range of cells you want to work with using the syntax `<sheetname!first-cell-name:last-cell-name>`. For example, use `Sheet1!A1:C3` to specify that you are working with the "Sheet1" tab, starting from cell "A1" to cell "C3". This syntax allows you to define specific ranges such as a row (`A1:A5`), a column (`A1:E1`), or a block (`A1:C3`) within the spreadsheet.
+
+### Writing Data to Cells
+
+For demonstration purposes, I will write simulated sensor data which includes a timestamp and sensor data.
+
+1. Drag the Inject node onto the canvas, and set `msg.payload` to `[$moment().format(), $random() * 100]` as a JSONata expression, and set it to repeat every 3 seconds of interval.
```

Comment:
Can we next time just inject and change node? For now this is fine, but not the `low-code` way.